vita3k/mem/src/mem.cpp

```diff
@@ -116,6 +116,17 @@ bool init(MemState &state) {
     return true;
 }
 
+Address alloc(MemState &state, size_t size, const char *name, unsigned int alignment) {
+    if (alignment != 0)
+        size += alignment - 1;
+    auto addr = alloc(state, size, name);
+    if (alignment == 0)
+        return addr;
+    auto new_addr = (addr + alignment - 1) & ~(alignment - 1);
+    state.aligned_addr_to_original[addr] = new_addr;
```

Comment:
Shouldn't this be the opposite?

Addr_to_original[new_addr]=addr ?

Reply:
Oh yes. :O